src/core/src/FabricMaterial.cpp

```diff
@@ -26,6 +26,51 @@ const auto DEFAULT_DISPLAY_OPACITY = 1.0;
 const auto DEFAULT_MATERIAL_INFO = GltfUtil::getDefaultMaterialInfo();
 const auto DEFAULT_TEXTURE_INFO = GltfUtil::getDefaultTextureInfo();
 const auto DEFAULT_TEXCOORD_INDEX = uint64_t(0);
+const auto DEFAULT_FEATURE_ID_PRIMVAR_NAME = std::string("_FEATURE_ID_0");
+const auto DEFAULT_NULL_FEATURE_ID = -1;
+
+struct FeatureIdCounts {
+    uint64_t indexCount;
+    uint64_t attributeCount;
+    uint64_t textureCount;
+    uint64_t totalCount;
+};
+
+FeatureIdCounts getFeatureIdCounts(const FabricMaterialDefinition& materialDefinition) {
+    const auto& featureIdTypes = materialDefinition.getFeatureIdTypes();
+    auto featureIdCount = featureIdTypes.size();
+
+    if (featureIdCount > FabricTokens::MAX_FEATURE_ID_COUNT) {
```

Comment:
Is 10 an arbitrary limit?

Reply:
Yeah kind of, let me see if the limit can be removed...

Reply:
I may do that in a follow up PR